core/arch/arm/plat-stm/platform_config.h

```diff
@@ -186,16 +186,9 @@
 /* Carvout out secure RAM range (emulated SRAM is expected near DRAM) */
 #if defined(CFG_WITH_PAGER) && defined(TZSRAM_BASE)
 #if TZSRAM_BASE >= CFG_DDR_START
-#if TZSRAM_BASE < TZDRAM_BASE
-#define STM_SECDDR_BASE		TZSRAM_BASE
-#else
-#define STM_SECDDR_BASE		TZDRAM_BASE
-#endif
-#if (TZSRAM_BASE + TZSRAM_SIZE) > (TZDRAM_BASE + TZDRAM_SIZE)
-#define STM_SECDDR_END		(TZSRAM_BASE + TZSRAM_SIZE)
-#else
-#define STM_SECDDR_END		(TZDRAM_BASE + TZDRAM_SIZE)
-#endif
+#define STM_SECDDR_BASE		MIN_RAW(TZSRAM_BASE, TZDRAM_BASE)
+#define STM_SECDDR_END		MAX_RAW(TZSRAM_BASE + TZSRAM_SIZE, \
+					TZDRAM_BASE + TZDRAM_SIZE)
```

Comment:
Why not `MIN()` and `MAX()`?

Reply:
This is the issue.
**platform_config.h** is included from both C ans ASM source files. On ASM file, `MIN()` & `MAX()` are ok, but from C source files,  `MIN()` & `MAX()` from **util.h** expects the arguments are typed (int, size_t, or whatever). Here, as the macros are used with raw value set, build complains:
```
In file included from core/arch/arm/include/arm.h:8:0,
                 from core/arch/arm/include/kernel/thread.h:11,
                 from core/arch/arm/kernel/asm-defines.c:7:
lib/libutils/ext/include/util.h:24:16: error: missing binary operator before token "("
  (__extension__({ __typeof__(a) _a = (a); \
                ^
core/arch/arm/plat-stm/./platform_config.h:190:25: note: in expansion of macro ‘MAX’
 #define STM_SECDDR_END  MAX(TZSRAM_BASE + TZSRAM_SIZE, \
                         ^~~
core/arch/arm/plat-stm/./platform_config.h:204:6: note: in expansion of macro ‘STM_SECDDR_END’
 #if (STM_SECDDR_END < 0x80000000ULL)
      ^~~~~~~~~~~~~~
make: *** [out/core/include/generated/.asm-defines.s] Error 1
```

Reply:
Ah, yes, sorry.

However I don't think the explanation in commit "libutils: introduce MAX_RAW() and MIN_RAW() for non-typed values" is correct. A simple test with `MIN(0x1234, 0x5678)` works fine, for example. TBH I don't understand what the compiler is complaining about.